src/ar_meta_db.erl

```diff
@@ -1,91 +1,179 @@
+%% @doc Defines a small in-memory metadata table for Arweave nodes.
+%% Typically used to store small peices of globally useful information
+%% (for example: the port number used by the node).
+%% @end
+
 -module(ar_meta_db).
--export([start/0, get/1, put/2, keys/0, remove_old/1,remove_old/2, increase/2]).
+-behaviour(gen_server).
+-define(SERVER, ?MODULE).
+
 -compile({no_auto_import, [{get, 1}, {put, 2}]}).
 -include_lib("eunit/include/eunit.hrl").
 -include("ar.hrl").
-%%% Defines a small in-memory metadata table for Arweave nodes.
-%%% Typically used to store small peices of globally useful information
-%%% (for example: the port number used by the node).
-
-%% @doc Initialise the metadata storage service.
-start() ->
-	spawn(
-		fun() ->
-			ar:report([starting_meta_db]),
-			ets:new(?MODULE, [set, public, named_table, {read_concurrency, true}]),
-			ets:new(blacklist, [set, public, named_table]),
-			receive stop -> ok end
-		end
-	),
-	% Add a short wait to ensure that the table has been created
-	% before returning.
-	receive after 250 -> ok end.
-
-%% @doc Put an Erlang term into the meta DB. Typically these are
-%% write-once values.
-put(Key, Val) -> ets:insert(?MODULE, {Key, Val}).
-%% @doc Retreive a term from the meta db.
+
+-export([get/1, put/2, keys/0, remove_old/1, increase/2]).
+-export([start_link/0, stop/0]).
+-export([init/1,
+	 handle_call/3,
+	 handle_cast/2,
+	 handle_info/2,
+	 terminate/2,
+	 code_change/3]).
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+%% @doc Insert key-value-pair into db
+put(Key, Value) ->
+	gen_server:call(?MODULE, {put, Key, Value}).
+
+%% @doc Retreive value for key
 get(Key) ->
-	case ets:lookup(?MODULE, Key) of
-		[{Key, Obj}] -> Obj;
-		[] -> not_found
-	end.
+	gen_server:call(?MODULE, {get, Key}).
 
-%% @doc Increase the value associated by a key by Val
+%% @doc Increase key with value
 increase(Key, Val) ->
-	put(Key, get(Key) + Val).
+	gen_server:call(?MODULE, {increase, Key, Val}).
 
 %% @doc Remove entries from the performance database older than
-%% ?PEER_TMEOUT
```

Comment:
Did you intend to remove this part of the docstring?

Reply:
Nope, good catch :+1: